### PR TITLE
mattdrayer/issue/2013: Allow site override in OrderCreator

### PR DESCRIPTION
**Pull request by mattdrayer**
Accompanying change set for Issue #2013 -- happy to revise.

Cheers,

Matt

### CODE REVIEW

File: src/oscar/apps/order/utils.py [hunk 1]
```diff
@@ -119,6 +118,8 @@ def create_order_model(self, user, basket, shipping_address,
             order_data['status'] = status
         if extra_order_fields:
             order_data.update(extra_order_fields)
+        if 'site' not in order_data:
+            order_data['site'] = Site._default_manager.get_current()
```
Comment (sasha0) [inline src/oscar/apps/order/utils.py]: Could be replaced with `order_data.setdefault('site', Site._default_manager.get_current())`
Original idea by @mvantellingen :wink:

**Comment by mvantellingen** [comment]
Thanks, can you also add a PR with a unittest for this? This makes sure we don't break it again in the future.

File: src/oscar/apps/order/utils.py (hunk 1 above)
Comment (mvantellingen) [inline src/oscar/apps/order/utils.py]: Using setdefault() here could cause an unnecessary call to get_current()

**Comment by sasha0** [comment]
@mvantellingen added tests in the PR #2080.

**Comment by mattdrayer** [comment]
Sorry guys, I have been meaning to get to this but time's been a bit tight due to a big project @ edX.

**Comment by sasha0** [comment]
@mattdrayer no worries - I suppose my tests will be enough for now.

**Pull request merged**